dask_expr/tests/test_string_accessor.py

```diff
@@ -98,3 +98,13 @@ def test_string_accessor(ser, dser, func, kwargs):
         )
     else:
         assert_eq(getattr(dser.index.str, func)(**kwargs), pdf_result)
+
+
+def test_str_accessor_not_available():
+    pdf = lib.DataFrame({"a": [1, 2, 3]})
+    df = from_pandas(pdf, npartitions=2)
+    # Not available on invalid dtypes
+    with pytest.raises(AttributeError, match=".str accessor"):
+        df.a.str
+
+    assert "str" not in dir(df.a)
```

Comment:
Should we also have one for `cat`?

Reply:
We have them in dask/dask, so didn't bother here